src/components/search/Filter/Filter.tsx

```diff
@@ -60,16 +92,30 @@ function Filter({
     }
   }, [screenWidth])
 
-  const onAccordionChange = (index: number) => {
-    if (indicesExpanded.has(index)) {
-      indicesExpanded.delete(index)
-      setIndicesExpanded(new Set(indicesExpanded))
+  // Ensures selected filters are up to date at opening
+  useEffect(() => {
+    !isOpen && setSelectedFilters(searchState.selectedFacets)
+  }, [isOpen, searchState.selectedFacets])
 
+  // Opens accordion items with active facets
+  useEffect(() => {
+    // Ensures all the active facets were identified
+    if (activeFacets.length !== filteredFacets.length) {
       return
     }
 
-    setIndicesExpanded(new Set(indicesExpanded.add(index)))
-  }
+    // Ensures there isn't empty facets
+    const filteredActiveFacets = activeFacets.filter(
+      (item) => item.facets.length > 0
+    )
```

Comment:
I'm confused regarding the concepts of filtered and active facets. Can you clarify?

Reply:
This filtering intends to prevent the array to have empty values (happens when there is no checked facets into the accordion item), so is basically to avoid unnecessary checks in empty arrays

Reply:
Got it. I think we can improve these names to clarify the code.

Regarding to facets, we have:
- `filteredFacets`
- `filteredActiveFacets`

These ones can be better named by telling what the filter does. i.e., if you're filtering the selected `activeFacets` the name should be `selectedActiveFacets`.